models/Form.php

```diff
@@ -0,0 +1,259 @@
+                        }
+                        ?>
+                    </div>
+                </div>
+                <?php
+            }else{
+                $model_article->edit($title,$content,$posted,$_GET['id']);
+             
+                if(!empty($_FILES['image']['name']))
+                {
+                    $model_article->update_img($_FILES['image']['tmp_name'], $extension);
+                    header("Location:/liste-de-tous-les-articles");
+                }
+                else
+                {
+                    header("Location:/liste-de-tous-les-articles");
+                }
+                
+                ?>
+                    <script>
+                        window.location.replace("index.php?page=postback&id=<?= $_GET['id'] ?>");
+                    </script> 
+                <?php
+            }
+        }
+    }
+    
+     // Fonction permettant de vérifier que les données envoyées pat l'utilisateur sont bonnes.
+    function form_comment_verification(){
+        if(filter_has_var(INPUT_POST, 'submit')){
+            $name = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'name', FILTER_SANITIZE_STRING)));
+            $email = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email', FILTER_SANITIZE_STRING)));
+            $comment = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'comment', FILTER_SANITIZE_STRING)));
+            $errors = [];
+
+            if(empty($name) || empty($email) || empty($comment)){
+                $errors['empty'] = "Tous les champs n'ont pas été remplis";
+            }else{
+                if(!filter_var($email, FILTER_VALIDATE_EMAIL)){
+                    $errors['email'] = "L'adresse email n'est pas valide";
+                }
+            }
+
+            if(!empty($errors)){
+            ?>
+                <div class="card red">
+                    <div class="card-content white-text">
+                        <?php
+                            foreach($errors as $error){
+                                echo $error;
+                            }
+                        ?>
+                    </div>
+                </div>
+            <?php
+            }else{
+                Comment::insert_comment($name,$email,$comment);
+            }
+        }
+    }
+    
+    function form_page_write(){
+        if(isset($_POST['post'])){
+            if(isset($_POST['title'])){
+                $title = htmlspecialchars(trim($_POST['title']));
+            }
+            if(isset($_POST['content'])){
+                $content = htmlspecialchars(trim($_POST['content']));
+            }
+
+            $posted = isset($_POST['public']) ? "1" : "0";
+
+            $errors = [];
+
+            if(empty($title) || empty($content)){
+                $errors['empty'] = "Veuillez remplir tous les champs";
+            }
+
+            if(!empty($_FILES['image']['name'])){
+                $file = $_FILES['image']['name'];
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Direct use of $_FILES Superglobal detected.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5344015)